File: frontend/src/api/shop.ts
import { api } from "./client";

export interface ShopSettings {
  id: string;
  name: string;
  owner_name: string | null;
  owner_phone: string | null;
  is_active: boolean;
  business_name: string | null;
  business_address: string | null;
  business_phone: string | null;
  business_email: string | null;
  business_upi: string | null;
  whatsapp_auto_send: boolean;
  whatsapp_enable_pdf: boolean;
  whatsapp_message_template: string | null;
  whatsapp_footer_message: string | null;
  whatsapp_language: string;
}

export interface ShopSettingsUpdate {
  business_name?: string | null;
  business_address?: string | null;
  business_phone?: string | null;
  business_email?: string | null;
  business_upi?: string | null;
  whatsapp_auto_send?: boolean | null;
  whatsapp_enable_pdf?: boolean | null;
  whatsapp_message_template?: string | null;
  whatsapp_footer_message?: string | null;
  whatsapp_language?: string | null;
}

export async function getShopSettings(): Promise<ShopSettings> {
  const { data } = await api.get<ShopSettings>("/shop");
  return data;
}

export async function updateShopSettings(payload: ShopSettingsUpdate): Promise<ShopSettings> {
  const { data } = await api.patch<ShopSettings>("/shop", payload);
  return data;
}

File: backend/app/schemas/shop.py
from __future__ import annotations
import uuid
from pydantic import BaseModel


class ShopSettingsOut(BaseModel):
    id: uuid.UUID
    name: str
    owner_name: str | None = None
    owner_phone: str | None = None
    is_active: bool
    business_name: str | None = None
    business_address: str | None = None
    business_phone: str | None = None
    business_email: str | None = None
    business_upi: str | None = None
    whatsapp_auto_send: bool
    whatsapp_enable_pdf: bool
    whatsapp_message_template: str | None = None
    whatsapp_footer_message: str | None = None
    whatsapp_language: str
    
    model_config = {"from_attributes": True}


class ShopSettingsUpdate(BaseModel):
    business_name: str | None = None
    business_address: str | None = None
    business_phone: str | None = None
    business_email: str | None = None
    business_upi: str | None = None
    whatsapp_auto_send: bool | None = None
    whatsapp_enable_pdf: bool | None = None
    whatsapp_message_template: str | None = None
    whatsapp_footer_message: str | None = None
    whatsapp_language: str | None = None
